source/daplink/drag-n-drop/vfs_user.c

```diff
@@ -517,22 +534,6 @@ static uint32_t update_details_txt_file(uint8_t *data, uint32_t datasize)
         pos += util_write_string(buf + pos, "\r\n");
     }
 
-#if defined(__CC_ARM)
-    pos += util_write_string(buf + pos, "Compiler: armcc\r\n");
-#elif (defined(__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050))
-    pos += util_write_string(buf + pos, "Compiler: armclang\r\n");
-#elif defined(__GNUC__)
-    pos += util_write_string(buf + pos, "Compiler: gcc\r\n");
-#endif
-
-    // GIT sha
-    pos += util_write_string(buf + pos, "Git SHA: ");
-    pos += util_write_string(buf + pos, GIT_COMMIT_SHA);
-    pos += util_write_string(buf + pos, "\r\n");
-    // Local modifications when making the build
-    pos += util_write_string(buf + pos, "Local Mods: ");
-    pos += util_write_uint32(buf + pos, GIT_LOCAL_MODS);
-    pos += util_write_string(buf + pos, "\r\n");
```

Comment:
As an alternative to compressing these fields to fit into 512 bytes, what are your thoughts on updating this function to support arbitrary length details.txt files? One possible way to do this would be to let DETAILS.TXT be written in chunks, so for each sector that needs to be read it is called with a different offset.

Example implementation of update_details_txt_file to allow read from arbitrary position:
```c
static uint32_t update_details_txt_file(uint32_t pos, uint8_t *buf, uint32_t datasize) {

    ...

    pos += util_write_string_in_region(buf, datasize, buf + pos, "Automation allowed: ");
    pos += util_write_string_in_region(buf, datasize, buf + pos, config_get_automation_allowed() ? "1" : "0");
    pos += util_write_string_in_region(buf, datasize, buf + pos, "\r\n");
    pos += util_write_string_in_region(buf, datasize, buf + pos, "Overflow detection: ");
    pos += util_write_string_in_region(buf, datasize, buf + pos, config_get_overflow_detect() ? "1" : "0");
    pos += util_write_string_in_region(buf, datasize, buf + pos, "\r\n");
    
    ...
}
```

DETAILS.TXT creation (buffer size of 10 rather than 512 used):
```
....Automa                                              ...
          tion allow                                    update_details_txt_file(200, buf, 10);
                    ed: 1\nOve                          update_details_txt_file(210, buf, 10);
                              rflow dete                update_details_txt_file(220, buf, 10);
                                        ction: 0..      ...
```

Reply:
I was thinking about something similar, but one problem is dealing with the `expand_info()` call. A `util_write_string_expand_in_region()` function would be tricky to implement without an additional buffer (e.g. `@R` is expanded to the URL that can contain other escape sequences like board ID or version).

Reply:
This would also eventually allow to remove `file_buffer` which is a `VFS_SECTOR_SIZE` (512 bytes) array that exists only to compute the size of the VFS files.

Reply:
For anything that needs to be expanded you could probably use a temporary stack buffer for the expansion and then copy the result with `util_write_string_in_region`

Reply:
I did an implementation of this idea in [3624cad](https://github.com/mbrossard/DAPLink/commit/3624cad4e0b916eeed7e4b14ffa3043a985c645e). It saves 512 bytes of RAM and as such takes ~568 bytes more Flash.